### DISCUSSION TITLE
Can mapper handle @property fields

### DISCUSSION THREAD

**Discussion by zadca123** [sqlalchemy/sqlalchemy#11780] [Usage Questions] [answered]
I don't want to use my private attributes, cuz they need to change a little. I didn't found any examples with private fields in the docs https://docs.sqlalchemy.org/en/20/orm/dataclasses.html#mapping-attrs-with-declarative-imperative-table. Or maybe i have stupid idea to trying to convert my current class to model. Should I just create `ThreadModel` or something
```
from __future__ import annotations
from attrs import define
from sqlalchemy import Column, Integer, String, Table
from sqlalchemy.orm import Mapped, registry

from sqlalchemy import create_engine
from sqlalchemy.orm import sessionmaker

DATABASE_URL = "sqlite:///:memory:"
engine = create_engine(DATABASE_URL, echo=True)
SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
mapper_registry = registry()


@mapper_registry.mapped
@define(slots=False, auto_attribs=True)
class Thread:
    __table__ = Table(
        "thread",
        mapper_registry.metadata,
        Column("number", Integer, primary_key=True),
        Column("board", String(10)),
    )
    _number: int
    _board: str

    def __attrs_post_init__(self):
        mapper_registry.metadata.create_all(engine)

    @property
    def number(self) -> Mapped[int]:
        return self._number + 10

    @property
    def board(self) -> Mapped[str]:
        return self._board + "qwerty"
```

`sqlalchemy.exc.ArgumentError: Type annotation for "Thread._number" can't be correctly interpreted for Annotated Declarative Table form.  ORM annotations should normally make use of the ``Mapped[]`` generic type, or other ORM-compatible generic type, as a container for the actual type, which indicates the intent that the attribute is mapped. Class variables that are not intended to be mapped by the ORM should use ClassVar[].  To allow Annotated Declarative to disregard legacy annotations which don't use Mapped[] to pass, set "__allow_unmapped__ = True" on the class or a superclass this class. (Background on this error at: https://sqlalche.me/e/20/zlpr)`

**Comment by CaselIT** [comment, marked as answer]
Hi,

First of all apply what the error says as solution to allow arbitrary types in orm classes.
Then if you want that `_number` be mapped to your `number` column in the table you have to follow the docs here https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#alternate-attribute-names-for-mapping-table-columns

Regarding property, the type annotations you are using are wrong, since they are accessed only in an instance of your class, operating on the values, not on the columns.
If you want to use `number` and `board` also in the queries you can look at https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.column_property

**Comment by zadca123** [comment]
Thank you,
Also sorry for not finding it on my own

**Reply by CaselIT** [reply]
no problem!
